File: app/src/main/java/com/zhan/doublepull/adapter/ScrollAdapter.java
package com.zhan.doublepull.adapter;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.zhan.doublepull.R;

public class ScrollAdapter extends RecyclerView.Adapter {

  private static final int TYPE_NORMAL = 1000;
  private static final int TYPE_HEADER = 2000;

  @NonNull
  @Override public RecyclerView.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
    if (viewType == TYPE_HEADER) {
      return new HeaderViewHolder(
          LayoutInflater.from(parent.getContext()).inflate(R.layout.header, parent, false));
    }
    return new ScrollViewHolder(
        LayoutInflater.from(parent.getContext()).inflate(R.layout.item, parent, false));
  }

  @Override public int getItemViewType(int position) {
    if (position == 0) {
      return TYPE_HEADER;
    }
    return TYPE_NORMAL;
  }

  @Override public void onBindViewHolder(@NonNull RecyclerView.ViewHolder holder, int position) {

  }

  @Override public int getItemCount() {
    return 31;
  }

  private class ScrollViewHolder extends RecyclerView.ViewHolder {

    public ScrollViewHolder(View itemView) {
      super(itemView);
    }
  }

  private class HeaderViewHolder extends RecyclerView.ViewHolder {

    public HeaderViewHolder(View itemView) {
      super(itemView);
    }
  }
}

File: app/src/main/java/com/zhan/doublepull/adapter/ScrollAdapter.kt
package com.zhan.doublepull.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView

import com.zhan.doublepull.R

class ScrollAdapter : RecyclerView.Adapter<RecyclerView.ViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
        return if (viewType == TYPE_HEADER) {
            HeaderViewHolder(
                    LayoutInflater.from(parent.context).inflate(R.layout.header, parent, false))
        } else ScrollViewHolder(
                LayoutInflater.from(parent.context).inflate(R.layout.item, parent, false))
    }

    override fun getItemViewType(position: Int): Int {
        return if (position == 0) {
            TYPE_HEADER
        } else TYPE_NORMAL
    }

    override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {

    }

    override fun getItemCount(): Int {
        return 31
    }

    private inner class ScrollViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView)

    private inner class HeaderViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView)

    companion object {

        private const val TYPE_NORMAL = 1000
        private const val TYPE_HEADER = 2000
    }
}
